### ISSUE TITLE
Mining started at height 1.

### ISSUE THREAD

**Issue by caowenyu75** [sammy007/open-ethereum-pool#301]
#### System information

**Geth version: `geth version` or commit hash if `develop`**
Geth
Version: 1.8.1-unstable
Git Commit: 9fd76e33af367752160ab0e33d1097e1e9aff6e4
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10
Operating System: linux
GOPATH=
GOROOT=/usr/local/go

**Redis version: `redis-server --version`**
Redis server v=4.0.8 sha=00000000:0 malloc=jemalloc-4.0.3 bits=64 build=3973c0fb8e1b686d

OS & Version: Linux/OSX

Commit branch and hash: (`master`, `ba60e1c`)

#### Expected behaviour
Sync finished and restart sync again.

#### Actual behaviour
INFO [02-24|02:14:05] Imported new block headers               count=1    elapsed=5.939ms   number=5147271 hash=3428bf…e1dbab ignored=0
INFO [02-24|02:14:08] Imported new block headers               count=1    elapsed=7.154ms   number=5147272 hash=04f08b…369747 ignored=0
WARN [02-24|02:14:11] Rolled back headers                      count=35   header=5147272->5147237 fast=7->7             block=7->7
WARN [02-24|02:14:43] Synchronisation failed, dropping peer    peer=e2c87a70876e581c err="retrieved hash chain is invalid"
INFO [02-24|02:14:46] Imported new block headers               count=0    elapsed=3.690ms   number=192     hash=723899…123390 ignored=192
INFO [02-24|02:14:46] Imported new block receipts              count=0    elapsed=23.318ms  number=2       hash=b495a1…4698c9 size=0.00B    ignored=2
INFO [02-24|02:14:46] Imported new block headers               count=0    elapsed=4.474ms   number=384     hash=d3d5d5…c79cf3 ignored=192
INFO [02-24|02:14:46] Imported new block headers               count=0    elapsed=6.607ms   number=576     hash=41a746…6a8b38 ignored=192
INFO [02-24|02:14:47] Imported new block headers          

#### Backtrace

````
[backtrace]
````
![sync issue](https://user-images.githubusercontent.com/7809689/36629390-2244df8a-1990-11e8-94cf-890740c27bd1.png)
basically, the mining always starts at height 1 once I restart the sync. on the mining side, the miner mines epoch 0.
![sync issue1](https://user-images.githubusercontent.com/7809689/36629501-74af4a16-1991-11e8-91be-3d6abb7aafac.png)

any ideas?

**Comment by caowenyu75** [comment]
basically, the mining always start at height 1 once I restart the sync. on the mining side, the miner mines epoch 0.
![default](https://user-images.githubusercontent.com/7809689/36629473-1143c9d4-1991-11e8-8920-f430bdb17a4f.png)

**Comment by caowenyu75** [comment]
i fixed it by sync with a full sync.
